api/v1beta1/types.go

```diff
@@ -799,6 +809,14 @@ type AzureBastion struct {
 	Subnet SubnetSpec `json:"subnet,omitempty"`
 	// +optional
 	PublicIP PublicIPSpec `json:"publicIP,omitempty"`
+	// BastionHostSkuName configures the tier of the Azure Bastion Host. Can be either Basic or Standard. Defaults to Basic.
+	// +kubebuilder:validation:Enum=Basic;Standard
+	// +optional
+	Sku BastionHostSkuName `json:"sku,omitempty"`
+	// EnableTunneling enables the native client support feature for the Azure Bastion Host. Defaults to false.
+	// +kubebuilder:default=false
+	// +optional
+	EnableTunneling bool `json:"enableTunneling,omitempty"`
```

Comment:
what do you think of adding the new fields to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/test/ci/prow-private/patches/bastion.yaml so we test them in our e2e test? And our test private clusters have tunneling enabled which might facilitate debugging (from my understanding it should not increase cost since the Azure default for the Sku is already Standard)